[DOC] Major documentation update

pvchupin opened the pull request:
* Simplified instructions for the first local run
* Clarified variables, parameters meaning
* Fixed inconsistencies with implementation
* Outlined special test categories External and ExtraTests
* Other minor fixes

TODO:
* Update LIT parameters, some are not used now, some are not documented

Signed-off-by: Pavel V Chupin <pavel.v.chupin@intel.com>

Commit:
[DOC] Major documenation rewrite

* Simplified instructions for the first local run
* Clarified variables, parameters meaning
* Fixed inconsistencies with implementation
* Outlined special test categories External and ExtraTests
* Other minor fixes

TODO:
* Update LIT parameters, some are not used now, some are not documented

Signed-off-by: Pavel V Chupin <pavel.v.chupin@intel.com>

vladimirlaz commented on SYCL/ExtraTests/README.md:
```diff
@@ -0,0 +1,29 @@
+"ExtraTests" directory contains infrastructure for picking up LIT style tests
+from another external directory passed in SYCL_EXTRA_TESTS_SRC.
+
+# Cmake parameters
```
```suggestion
# CMake parameters
```

vladimirlaz committed:
Update SYCL/ExtraTests/README.md

bader commented on SYCL/External/README.md:
```diff
@@ -1,82 +1,75 @@
+application/test.
 
 # Directory structure
 Every sub-directory (e.g. RSBench) contains the following content:
```
What is `RSBench`? Can we add a link to this benchmark?

pvchupin replied:
Sure. Added.

bader commented on SYCL/External/README.md:
```diff
@@ -1,82 +1,75 @@
- used by it (see [Main parameters](#main-parameters) section for details)
+ application binary and data files.
  * **\*.test** - test configuration files, containing command lines, test
-   status lines and verification patterns following [LLVM test infrastructure](https://llvm.org/docs/TestingGuide.html).
+   status lines and verification patterns following
+   [LLVM tests infrastructure](https://llvm.org/docs/TestingGuide.html).
```
Why is this required? My understanding is that only **lit.local.cfg** is required and this file configures now external benchmark is tested. As you can see below the file name extension is defined by this file and can be something different (e.g. `*.cpp`).

pvchupin replied:
I think you are right in general and probably depends on case. It's just how RSBench example has been integrated. This approach is also documented at https://llvm.org/docs/TestingGuide.html#platform-specific-tests.
I've rephrased this section to focus more on RSBench as an example, rather than set of real requirements. Other cases may or may not needed to reuse these.

pvchupin replied:
Also I'm not sure if RSBench test is functional at all. It looks it's marked as XFAIL for all platforms. I don't even see it downloading sources when I tried to run on host. @vladimirlaz, can you comment on that? Is that a placeholder of some sort?

bader commented on SYCL/ExtraTests/README.md:
```diff
@@ -0,0 +1,28 @@
+# Overview
+
+"ExtraTests" directory contains infrastructure for picking up LIT style tests
+from another external directory passed in SYCL_EXTRA_TESTS_SRC.
```
Could you clarify what is the difference with [External](https://github.com/intel/llvm-test-suite/tree/intel/SYCL/External) tests, please?
It looks to me that both are intended to run LIT-based tests located out-of-tree. Right?

pvchupin replied:
It seems ExtraTests run tests which are completely under LIT framework (e.g. in directory elsewhere) and External tests are to run other frameworks. That's my guess.
@vladimirlaz, can you confirm? What was the purpose and do we use any of these right now?

vladimirlaz replied:
@pvchupin you are right about ExtraTests and External test.
 - ExtraTests can be used to run intel/llvm in-tree LIT tests from llvm-test-suite framework. This can be removed I think.
 - External tests allow to download/build/run external applications (e.g. RSBench from https://github.com/ANL-CESAR/RSBench.git). The tests were marked as XFail because benchmark internal data verification failed (it is successfully build and code is executed without crashes). There is no current need for that as well.

pvchupin replied:
Marked both as experimental. We should probably remove at some later point if no use.

bader commented on SYCL/README.md:
```diff
@@ -1,175 +1,167 @@
+# Prerequisites
+
+ - DPC++ compiler. Can be built following these
+   [instructions](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#build-dpc-toolchain)
```
Shouldn't we recommend to take pre-built binaries instead of asking to build it from sources? E.g. Contributing guide describes CI checks taking nightly builds.

pvchupin replied:
Updated text.

Commit:
Addressing review comments

Also found that -DCHECK_SYCL_ALL is really required for the first run
-DSYCL_TARGET_DEVICES alone is not working.

TODO: need to clarify External/ExtraTests functionality

Commit:
Marking External and ExtraTests as experimental

Commit:
Fixing formatting

Commit:
One more time
